src/server.ts

```diff
@@ -110,19 +123,48 @@ export function getViewerFullUrl(
   let viewerParams = `src=${escapeParam(sourceUrl.href)}`;
   viewerParams += `&bookMode=${!singleDoc}&renderAllPages=${!quick}`;
 
+  let viewerCustomStyleParam: 'style' | 'userStyle' = 'style';
+  let viewerCustomStyle = '';
+
   if (style) {
-    viewerParams += `&style=${escapeParam(style)}`;
+    if (isCssString(style)) {
+      viewerCustomStyle = style;
+    } else {
+      viewerParams += `&style=${escapeParam(style)}`;
+    }
   }
 
   if (userStyle) {
-    viewerParams += `&userStyle=${escapeParam(userStyle)}`;
+    if (isCssString(userStyle)) {
+      if (!viewerCustomStyle) {
+        viewerCustomStyle = userStyle;
+        viewerCustomStyleParam = 'userStyle';
+      } else {
+        viewerParams += `&userStyle=data:,${encodeURIComponent(userStyle)}`;
+      }
+    } else {
+      viewerParams += `&userStyle=${escapeParam(userStyle)}`;
+    }
   }
 
-  if (pageSizeValue) {
-    viewerParams +=
-      '&userStyle=data:,/*<viewer>*/' +
-      encodeURIComponent(`@page{size:${pageSizeValue}!important;}`) +
-      '/*</viewer>*/';
+  if (viewerCustomStyle || pageSizeValue || cropMarks || bleed || cropOffset) {
+    let pageStyle = '@page{';
+    if (pageSizeValue) {
+      pageStyle += `size:${pageSizeValue}!important;`;
+    }
+    if (cropMarks) {
+      pageStyle += `marks:crop cross!important;`;
+    }
+    if (bleed || cropMarks) {
+      pageStyle += `bleed:${bleed ?? '3mm'}!important;`;
+    }
+    if (cropOffset) {
+      pageStyle += `crop-offset:${cropOffset}!important;`;
+    }
+    pageStyle += '}';
+    viewerParams += `&${viewerCustomStyleParam}=data:,/*<viewer>*/${encodeURIComponent(
+      pageStyle,
+    )}/*</viewer>*/${encodeURIComponent(viewerCustomStyle)}`;
```

Comment:
> The CSS code is appended after the `/*<viewer>*/…/*</viewer>*/` block in the custom style settings Viewer parameter so that the CSS code is shown in the Edit CSS box in the Settings panel of the Viewer.

Can you add this as a comment for the later code maintainer?